File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        left , right , max_area = 0 , len(height)-1 , 0
        while left < right:
            max_area=max(max_area,min(height[left],height[right])*(right-left))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return max_area

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int l = 0, r = n-1;
        int mxArea = 0;
        while(l<r){
            int area = min(height[l], height[r]) * abs(l-r);
            mxArea = max(area, mxArea);
            if(height[l]<height[r]) l++;
            else r--;
        }
        return mxArea;
    }
};